abn/application/application.go

```diff
@@ -0,0 +1,116 @@
+package application
+
+// application.go - type of application
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/iter8-tools/iter8/base/log"
+)
+
+// Application is an application observed in a kubernetes cluster
+type Application struct {
+	// Name is of the form namespace/name where the name is the value of the label app.kubernetes.io/name
+	Name string `json:"name" yaml:"name"`
+	// Tracks is map from application track identifier to version name
+	Tracks `json:"tracks" yaml:"tracks"`
+	// Versions is a map of versions name to version data
+	Versions `json:"versions" yaml:"versions"`
```

Comment:
// Map each version to its metrics (version data = metrics)

Reply:
comment improved